### PR TITLE
Allow showing partial tiles on the edges of an image or map.

### PR DESCRIPTION
**Pull request by manthey**
If the tileLayer tilesMaxBounds function is defined, some tiles may be cropped.  Tile cropping is implemented in the gl, canvas, and null renderers (but not d3).

This fixes issue #544.

### CODE REVIEW

File: src/gl/quadFeature.js [hunk 1]
```diff
@@ -35,12 +35,25 @@ var gl_quadFeature = function (arg) {
...
       m_quads;
+  var fragmentShaderImageSource = [
+    'varying highp vec2 iTextureCoord;',
+    'uniform sampler2D sampler2d;',
+    'uniform mediump float opacity;',
+    'uniform highp vec2 crop;',
+    'void main(void) {',
+    '  mediump vec4 color = texture2D(sampler2d, iTextureCoord);',
+    '  if (iTextureCoord.s > crop.s || 1.0 - iTextureCoord.t > crop.t) {',
+    '    return;',
```
Comment (aashish24) [inline src/gl/quadFeature.js]: I think discard would be a better option here.

**Comment by aashish24** [comment]
@manthey so this is only going to crop towards right and bottom? What happens to the cropping towards left and top?

**Comment by manthey** [comment]
@aashish24 Yes, currently all tile sets I know of have the 0, 0 pixel within a tile, so only right and bottom cropping is necessary.  All partial-tile examples I have are from medical images, and this satisfies the need there.  If you have a real-world tile set which has partial tiles on the top or left, then we can add support at that time.

File: src/gl/quadFeature.js (hunk 1 above)
Comment (manthey) [inline src/gl/quadFeature.js]: Thanks.

**Pull request merged**